src/main/java/seedu/address/logic/CommandList.java

```diff
@@ -0,0 +1,52 @@
+package seedu.address.logic;
+
+import java.util.ArrayList;
+
+/**
+ * Records the latest user inputs.
+ * Only the latest MAX_RECORD_NUMBER inputs will be recorded.
+ */
+public class CommandList {
+    private static final int MAX_RECORD_NUMBER = 3;
+    private static final ArrayList<String> inputHistory = new ArrayList<>();
+
+    /**
+     * Records the latest user input.
+     * @param lastCommand the last user input.
+     */
+    public static void record(String lastCommand) {
+        if (!lastCommand.equals("history")) {
+            inputHistory.add(lastCommand);
+        }
+        if (inputHistory.size() > MAX_RECORD_NUMBER) {
+            clearOldCommand();
+        }
+    }
+
+    /** Gets the latest user input. */
+    public static String getLastCommand() {
+        if (inputHistory.size() == 0) {
+            return "";
+        }
+        return inputHistory.get(inputHistory.size() - 1);
+    }
+
+    /** Gets the most recent commands(number capped at MAX_RECORD_NUMBER). */
+    public static String getRecentCommands() {
+        StringBuilder result = new StringBuilder(inputHistory.get(0));
```

Comment:
Shouldn't this be inputHistory.get(inputHistory.size() - 1)? Since you want the most recent command first?